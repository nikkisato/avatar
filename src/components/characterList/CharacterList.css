
ul {
    display: flex;
    flex-direction: row;
    margin-left: 50px;
    flex-wrap: wrap;
    text-align:Center;
    
  }
li {
    list-style-type: none;   
     text-align:Center;
    


}

img{
    width: 300px;
    height: 300px;
}