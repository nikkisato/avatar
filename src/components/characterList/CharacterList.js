import React from 'react';
import Character from '../character/Character';
import PropTypes from 'prop-types';
import { useAvatarCharacter } from '../../hooks/AvatarHook';

const CharacterList = () => {
  const { character } = useAvatarCharacter();
  const CharacterElements = character.map(character => (
    <li key={character.id}>
      <Character />
    </li>
  ));
  return <ul>{CharacterElements}</ul>;
};

CharacterList.prototype = {
  characters: PropTypes.array.isRequired
};

export default CharacterList;
