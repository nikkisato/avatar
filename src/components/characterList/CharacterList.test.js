import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  it('renders CharacterList', () => {
    const wrapper = shallow(<CharacterList />);
    expect(wrapper).toMatchSnapshot();
  });
});
