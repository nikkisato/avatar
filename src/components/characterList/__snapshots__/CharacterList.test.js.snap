// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<div
  className="flex"
>
  <button
    onClick={[Function]}
  >
    Prev
  </button>
  ;
  <button
    onClick={[Function]}
  >
    Next
  </button>
  ;
  <ul />
</div>
`;
